tapir/coop/services/MembershipResignationService.py

```diff
@@ -78,6 +78,15 @@ def delete_end_dates(member: MembershipResignation):
         )
 
     @staticmethod
-    @transaction.atomic
     def delete_shareowner_membershippauses(resignation: MembershipResignation):
-        MembershipPause.objects.filter(share_owner=resignation.share_owner).delete()
+        for pause in MembershipPause.objects.filter(share_owner=resignation.share_owner):
+            if resignation.pay_out_day is not None:
+                if resignation.pay_out_day >= pause.end_date:
```

Comment:
Also we need to handle the cases when the pause doesn't have an end. If we compare a date with None we'll get an error.